ext/drsyms/drsyms_unix_common.c

```diff
@@ -439,6 +460,114 @@ addrsearch_symtab(dbg_module_t *mod, size_t modoffs, drsym_info_t *info INOUT, u
     return drsym_obj_symbol_offs(mod->obj_info, idx, &info->start_offs, &info->end_offs);
 }
 
+/******************************************************************************
+ * Hashtable building for symbol lookup.
+ *
+ * We use __wrap_malloc because our demangled strings have larger capacities
+ * than strlen() will find.
```

Comment:
I don't understand this. __wrap_malloc stores the size in the allocated block, but I don't see you using it anywhere. Nor calling strlen() on anything but the original symbol names, which are not allocated here.

Reply:
It's the drsym_demangle_helper code which has a loop to find the right size for the demangled symbol name, starting at 1024.  So that 1024 may well be much larger than strlen.  If it weren't for that over-capacity we would not need the stored size and could use strlen for the size on all frees.

Reply:
We could also have the hashtable strdup for us but that seems too wasteful when we're already allocating separately for all but the original name.

Reply:
I elaborated in the comment to help make this more clear.